server/controllers/finance/reports/priceList/index.js

```diff
@@ -0,0 +1,109 @@
+ *
+ * @requires Employee
+ * @requires ReportManager
+ */
+const _ = require('lodash');
+const ReportManager = require('../../../../lib/ReportManager');
+const db = require('../../../../lib/db');
+const util = require('../../../../lib/util');
+
+
+const TEMPLATE = './server/controllers/finance/reports/priceList/report.handlebars';
+
+const PDF_OPTIONS = {
+  filename : 'FORM.LABELS.PRICE_LIST',
+};
+
+exports.report = (req, res, next) => {
+  const options = req.query;
+
+  let report;
+
+  _.defaults(options, PDF_OPTIONS);
+
+  // set up the report with report manager
+  try {
+    report = new ReportManager(TEMPLATE, req.session, options);
+
+    const priceListUUID = db.bid(req.params.uuid);
+    return lookupPriceList(priceListUUID).then(data => {
+      const { items } = data;
+
+      items.forEach(item => {
+        if (item.priceListValue) {
+          if (item.is_percentage === 1) {
+            item.price += util.roundDecimal((item.price * item.priceListValue) / 100, 2);
+          } else {
+            item.price += util.roundDecimal(item.priceListValue, 2);
+          }
+        }
+      });
+      // group by inventory group
+      let groups = _.groupBy(items, i => i.groupName);
+
+      // make sure that they keys are sorted in alphabetical order
+      groups = _.mapValues(groups, lines => {
+        _.sortBy(lines, 'label');
+        return lines;
+      });
+      return report.render({ groups });
+    })
+      .then(result => {
+        res.set(result.headers).send(result.report);
+      })
+      .catch(next)
+      .done();
+
+  } catch (e) {
+    return next(e);
+  }
+};
+
+
+function lookupPriceList(uuid) {
+  let _priceList;
+  let sql =
+    `SELECT BUID(uuid) AS uuid, label, description, created_at, updated_at
+    FROM price_list WHERE uuid = ?;`;
+
+  return db.one(sql, [uuid])
+    .then((row) => {
+      _priceList = row;
+
+      sql = `
+      SELECT BUID(inventory.uuid) as uuid, inventory.code, inventory.text AS label, inventory.price, 
+        iu.abbr AS unit,
+        it.text AS type, ig.name AS groupName, BUID(ig.uuid) AS group_uuid, ig.expires, 
+        ig.unique_item, inventory.consumable,inventory.locked, inventory.stock_min,
+        inventory.stock_max, inventory.created_at AS timestamp, inventory.type_id, inventory.unit_id,
+        inventory.note,  inventory.unit_weight, inventory.unit_volume, 
+        ig.sales_account, ig.stock_account, ig.donation_account, inventory.sellable, inventory.note, 
+        inventory.unit_weight, inventory.unit_volume, ig.sales_account, ig.stock_account, ig.donation_account,
+        ig.cogs_account, inventory.default_quantity, s1.priceListValue, s1.is_percentage
+      FROM inventory JOIN inventory_type AS it
+        JOIN inventory_unit AS iu JOIN inventory_group AS ig ON
+        inventory.type_id = it.id AND inventory.group_uuid = ig.uuid AND
+        inventory.unit_id = iu.id
```

Comment:
Could you write JOINs like this : 
```sql
FROM inventory 
JOIN inventory_group ig ON ig.uuid = inventory.group_uuid
JOIN inventory_type it ON it.id = inventory.type_id
JOIN inventroy_unit iu ON iu.id = inventory.unit_id
```
It is more clear to read

Reply:
I used on of sql query in the stock core